src/librustc/hir/map/def_collector.rs

```diff
@@ -27,7 +27,7 @@ pub struct DefCollector<'a> {
     hir_crate: Option<&'a hir::Crate>,
     definitions: &'a mut Definitions,
     parent_def: Option<DefIndex>,
-    pub visit_macro_invoc: Option<&'a mut FnMut(NodeId, DefIndex)>,
+    pub visit_macro_invoc: Option<&'a mut FnMut(NodeId, DefIndex, bool /* const_integer */)>,
```

Comment:
I was going to say it would be better to have a struct than a tuple at this point (esp with a `bool` with comment). However, given that this field only ever seems to have one instantiation, it might just be over-abstracted. Could it be changed to a flag with a method rather than an optional closure?